linera-base/src/identifiers.rs

```diff
@@ -158,6 +199,49 @@ impl<A> Clone for ApplicationId<A> {
 
 impl<A> Copy for ApplicationId<A> {}
 
+impl<A: std::cmp::PartialEq> PartialEq for ApplicationId<A> {
+    fn eq(&self, other: &Self) -> bool {
+        self.bytecode_id == other.bytecode_id && self.creation == other.creation
```

Comment:
We could pattern match one of them, so that in case we later add a field the compiler will tell us to add it to the `PartialEq` implementation. Same for the `Hash` impl.

```suggestion
        let ApplicationId {
            bytecode_id,
            creation,
        } = other;
        self.bytecode_id == bytecode_id && self.creation == creation
```

Reply:
Great idea! Will do